crates/typst-cli/src/world.rs

```diff
@@ -187,17 +195,21 @@ impl World for SystemWorld {
     }
 
     fn today(&self, offset: Option<i64>) -> Option<Datetime> {
-        let now = self.now.get_or_init(chrono::Local::now);
+        let now = match &self.now {
+            Now::Fixed(time) => time,
+            Now::System(time) => time.get_or_init(Utc::now),
+        };
 
-        let naive = match offset {
+        // The time with the specified UTC offset, or within the local time zone.
```

Comment:
I don't think that's what happens. `now` is a `DateTime<Utc>`, there's nowhere that a local time zone could come from (`naive_local()` returns the same thing as `naive_utc()`, because the timezone *is* `Utc`). Previously, `now` was a `DateTime<Local>`, which does have local timezone information.

Reply:
Very good catch! It should behave correctly now.
I also noticed that leap-seconds were not preserved if an offset was given. I fixed that too.